core/Azure.Mcp.Core/src/Areas/Server/Commands/ServiceStartCommand.cs

```diff
@@ -734,4 +739,91 @@ private static WebApplication UseHttpsRedirectionIfEnabled(WebApplication app)
 
         return app;
     }
+
+    /// <summary>
+    /// Configures OpenTelemetry tracing for self-hosted HTTP mode with Azure Monitor exporter.
+    /// </summary>
+    /// <param name="options">The server configuration options.</param>
+    /// <returns>
+    /// A <see cref="TracerProvider"/> instance if telemetry is enabled and properly configured for HTTP transport;
+    /// otherwise, <c>null</c>.
+    /// </returns>
+    /// <remarks>
+    /// Telemetry is only configured when:
+    /// <list type="bullet">
+    /// <item><description>The transport is HTTP (not STDIO)</description></item>
+    /// <item><description>AZURE_MCP_COLLECT_TELEMETRY is not explicitly set to false</description></item>
+    /// <item><description>APPLICATIONINSIGHTS_CONNECTION_STRING environment variable is set</description></item>
+    /// </list>
+    /// The tracer provider includes ASP.NET Core and HttpClient instrumentation with filtering
+    /// to avoid duplicate spans and telemetry loops.
+    /// This telemetry configuration is intended for self-hosted scenarios where
+    /// the MCP server is running in HTTP mode. This creates an independent telemetry pipeline using TracerProvider to export
+    /// traces to user-configured Application Insights instance only when the necessary environment variables are set. This also honors 
+    /// the AZURE_MCP_COLLECT_TELEMETRY environment variable to allow users to disable telemetry collection if desired. Note that this is 
+    /// in addition to the telemetry configured in <see cref="OpenTelemetryExtensions"/>.
+    /// </remarks>
+    private static TracerProvider? ConfigureSelfHostingTelemetry(ServiceStartOptions options)
```

Comment:
Nit: for the method name and the description, we're very narrowly only adding the spans for incoming and outgoing HTTP requests. All other spans, logs, and metrics are elsewhere. So the name `ConfigureSelfHostingTelemetry` is inaccurate.